### PR TITLE
Feature/146 stream location bundle

### PR DESCRIPTION
**Pull request by dlin2102**
I wasn't sure what the live url was so I just used "/water/locations/streams/:location_code" as a placeholder.

### CODE REVIEW

**Comment by brian428** [comment]
I updated the REST url and went ahead and try to use the locations in the UI. Let me know if you see any possible downside to actually using them in the dropdown.

**Comment by dlin2102** [comment]
Can't think of any. I think it should be fine.

File: src/app-pages/map/components/location-stream-controls/LocationStreamControls.js [hunk 1]
```diff
@@ -31,6 +34,9 @@ const LocationStreamControls = ({ locationDetailData, fullScreen, doLocationsMap
     justifyContent: "space-between",
   };
 
+  // Don't show stream controls if there were no stream locations found
+  if( options.length === 0 ) return <></>
```
Comment (mohitb3) [inline src/app-pages/map/components/location-stream-controls/LocationStreamControls.js]: If streamLocations is an empty array, then the "Back to map" link doesn't display on the location page

**Comment by mohitb3** [comment]
I simplified the code by calling doStreamLocationsFetch directly in the LocationStreamControls component. When I tested it against the live endpoint, almost all the locations return an empty array for the stream locations.

**Review by mohitb3 — APPROVED** [review]

**Comment by brian428** [comment]
@mohitb3 I think that's expected. I ran a query and it looks like the stream locations table has about 2800 locations in it, which is much less than the 10,000 or so locations we pull in for our location summary. So basically it looks like only a fairly small subset of locations are part of a "stream".

**Comment by brian428** [comment]
I pushed an update to use the right ID for the stream locations. Also added logic to select the current location in the stream location list. And last, tweaked the rest bundle so we can use reduceFurther on any action, so that we can more easily clear out old data before running another load. Up til now, if there was an error or a race condition (like the stream location load starting before the new detail location code is set), the outdated data would still be maintained in the state.

https://github.com/USACE/water-ui/pull/162/commits/ac49391ee3b72754f8ec22238a840282f5bf6ffd

**Comment by brian428** [comment]
@mohitb3 and @dlin2102 let me know if you see any issue with my last commit. If it looks good I'll merge it.

**Comment by dlin2102** [comment]
Looks good to me.

**Pull request merged**